addons/spreadsheet/static/src/data_sources/display_name_repository.js

```diff
@@ -0,0 +1,200 @@
+        this.dataFetchedCallback = whenDataIsFetched;
+        /**
+         * Contains the display names of records. It's organized in the following way:
+         * {
+         *     "res.country": {
+         *         1: {
+         *              "value": "Belgium",
+         *              "deferred": Deferred<"Belgium">,
+         *     },
+         * }
+         */
+        /** @type {Object.<string, Object.<number, DisplayNameResult>>}*/
+        this._displayNames = {};
+        this._orm = orm;
+        this._endpoints = {};
+    }
+
+    /**
+     * Get the display name of the given record.
+     *
+     * @param {string} model
+     * @param {number} id
+     * @returns {Deferred<string|undefined>}
+     */
+    async waitForDisplayName(model, id) {
+        const displayNameResult = this._displayNames[model] && this._displayNames[model][id];
+        if (!displayNameResult) {
+            return this._fetchDisplayName(model, id);
+        }
+        return displayNameResult.deferred;
+    }
+
+    /**
+     * Set the display name of the given record. This will prevent the display name
+     * from being fetched in the background.
+     *
+     * @param {string} model
+     * @param {number} id
+     * @param {string} displayName
+     */
+    setDisplayName(model, id, displayName) {
+        if (!this._displayNames[model]) {
+            this._displayNames[model] = {};
+        }
+        this._displayNames[model][id] = {
+            state: "COMPLETED",
+            deferred: Promise.resolve(displayName),
+            value: displayName,
+        };
+    }
+
+    /**
+     * Get the display name of the given record. If the record does not exist,
+     * it will return undefined and fetch the display name in the background.
+     *
+     * @param {string} model
+     * @param {number} id
+     * @returns {string|undefined}
+     */
+    getDisplayName(model, id) {
+        const displayNameResult = this._displayNames[model] && this._displayNames[model][id];
+        if (!displayNameResult) {
+            // Catch the error to prevent the error from being thrown in the
+            // background.
+            this._fetchDisplayName(model, id).catch(() => {});
+            throw new LoadingDataError();
+        }
+        switch (displayNameResult.state) {
+            case "ERROR":
+                throw displayNameResult.error;
+            case "COMPLETED":
+                return displayNameResult.value;
+            default:
+                throw new LoadingDataError();
+        }
+    }
+
+    /**
+     * Get the batch endpoint for the given model. If it does not exist, it will
+     * be created.
+     *
+     * @param {string} model
+     * @returns {BatchEndpoint}
+     */
+    _getEndpoint(model) {
+        if (!this._endpoints[model]) {
+            this._endpoints[model] = new BatchEndpoint(this._orm, model, "name_get", {
+                whenDataIsFetched: () => this.dataFetchedCallback(),
```

Comment:
don't think it requires the anonymous casting:
whenDataIsFetched:  this.dataFetchedCallback  should suffice ?

Reply:
for the sake of writing the same thing for all  arguments

Reply:
Nope, `dataFetchedCallback` use the `this` of the caller. If we do that, we will lose the binding. What we could do is `whenDataIsFetched: this.dataFetchedCallback.bind(this)`, but not sure it's better